encoder/vaapiencoder_vp8.cpp

```diff
@@ -75,6 +75,10 @@ void VaapiEncoderVP8::resetParams()
 {
     //5 times compress ratio
     m_maxCodedbufSize = width() * height() * 3 / 2;
+    //when the resolution of video is very small such as 96x96 and buffer plus the header of vp8,
+    //the size of actual output buffer maybe be bigger the m_maxCodedbufSize.
+    //So plusing 4096 avoids the situation.
+    m_maxCodedbufSize += 0x1000;
```

Comment:
Yes, But it is only an experience value for him, 48x188 should have same problem, and it may related to the content. If the content is complex, it should have problem for high resolution.


From: Daniel Charles [mailto:notifications@github.com]
Sent: Saturday, July 2, 2016 12:20 AM
To: 01org/libyami <libyami@noreply.github.com>
Cc: Xu, Guangxin <guangxin.xu@intel.com>; Comment <comment@noreply.github.com>
Subject: Re: [01org/libyami] encoder/vp8: fix small resolution video encoding bug (#486)


In encoder/vaapiencoder_vp8.cpp<https://github.com/01org/libyami/pull/486#discussion_r69323788>:

> @@ -75,6 +75,10 @@ void VaapiEncoderVP8::resetParams()

>  {

>      //5 times compress ratio

>      m_maxCodedbufSize = width() * height() * 3 / 2;

> +    //when the resolution of video is very small such as 96x96 and buffer plus the header of vp8,

> +    //the size of actual output buffer maybe be bigger the m_maxCodedbufSize.

> +    //So plusing 4096 avoids the situation.

> +    m_maxCodedbufSize += 0x1000;

@clearlylin<https://github.com/clearlylin> already explained that the maximum resolution needing the padding is 96x96. Based on that information the if can be included.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/01org/libyami/pull/486/files/d8ace19d20da9eb807147ec9cb73668de1de7868#r69323788>, or mute the thread<https://github.com/notifications/unsubscribe/AGW9Gqo7qnEVgMUD-Tro2_nIwxa85tmnks5qRT4_gaJpZM4I8YpG>.